samples/ordered_puzzle/components/ordered_puzzle.js

```diff
@@ -0,0 +1,94 @@
+AFRAME.registerComponent("ordered-puzzle", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+
+  multiple: true,
+
+  init: function () {
+    const data = this.data.jsonData;
+    const el = this.el;
+
+    const numPuzzlePieces = data.images.length;
+
+    for (i = 0; i < numPuzzlePieces; i++) {
+      var textBoxProp =
+        '{"width": "' +
+        data.images[i].width +
+        '", "height": "' +
+        data.images[i].height +
+        '", "color": "yellow", "x": "' +
+        data.images[i].xTarget +
+        '", "y": "' +
+        data.images[i].yTarget +
+        '", "z": "0", "text": "' +
```

Comment:
Is there a reason for text: i here? Is that being used in target-box?

Reply:
It's currently not being used in target-box, but I was thinking in the future if we wanted to include text on the target boxes that could also be an option. I'll remove it in the PR since it's not being used.

Reply:
Ah I see, that makes sense. Feel free to keep this in here. SInce there is a lot of overlap between target-box and text-box, when integrating we will just use text-box with an if statement for adding text if `text != ""`.